app/src/main/java/com/example/kitchenflow/ui/main/OrdersAdapter.kt

```diff
@@ -53,6 +55,43 @@ class OrdersAdapter(
         }
     }
 
+    private fun ItemOrderBinding.setUpTimer(order: OrderModel) {
+        val currentTime = System.currentTimeMillis()
+        var orderPrepTime =
+            order.scheduledFor.convertToMilliseconds() - order.preparationTimeSec * 1000 + KITCHEN_PREP_MILLIS
+        if (order.orderType == OrderType.Curbside || order.orderType == OrderType.Takeout) {
+            ioTimerTv.text = if (currentTime >= orderPrepTime) {
+                ioTimerTv.setTextColor(context.getColor(R.color.colorPrimaryRed))
+                order.timer = ((currentTime - orderPrepTime) / 1000 / 60).toInt()
+                "+${if (order.timer >= 99) 99 else order.timer}m"
+            } else {
+                "${((orderPrepTime - currentTime) / 1000 / 60).toInt()}m"
+            }
+        } else {
+            orderPrepTime -= MIN_DRIVE_TIME_MILLIS
+            ioTimerTv.text = if (currentTime >= orderPrepTime) {
+                ioTimerTv.setTextColor(context.getColor(R.color.colorPrimaryRed))
+                order.timer = ((currentTime - orderPrepTime) / 1000 / 60).toInt()
```

Comment:
check `java.util.concurrent.TimeUnit` it is much more readable to use TimeUnit.MINUTES.toMillis(1) for example